schism/compiler.ss

```diff
@@ -178,28 +181,40 @@
   (define (read-library-from-file filename)
     (%open-as-stdin filename)
     (read))
-  
+
+  (define (find-library-file name)
+    (let ((search-paths '("./test/lib"
+                          "./lib"
+                          "./scheme-lib"))
+          (path-suffix (string-append
+                        (fold-left (lambda (path part)
+                                     (string-append
+                                      path
+                                      (string-append "/" (symbol->string part))))
+                                   ""
+                                   name)
+                        ".ss")))
+      (fold-left (lambda (found base-path)
```

Comment:
Here it might be nice to define an `or-map`; YMMV though :)

Reply:
Good idea! I think I'll save this for a future cleanup though.